moveit_kinematics/ikfast_kinematics_plugin/scripts/create_ikfast_moveit_plugin.py

```diff
@@ -50,284 +53,294 @@
 import yaml
 from lxml import etree
 import shutil
+import argparse
 
 plugin_gen_pkg = 'moveit_kinematics'  # package containing this file
 plugin_sub_dir = 'ikfast_kinematics_plugin' # sub directory which contains the template directory
 # Allowed search modes, see SEARCH_MODE enum in template file
 search_modes = ['OPTIMIZE_MAX_JOINT', 'OPTIMIZE_FREE_JOINT' ]
 
+
+def createParser():
+  parser = argparse.ArgumentParser(description="Generate an IKFast MoveIt! kinematic plugin")
+  parser.add_argument("robot_name", help="The name of your robot")
+  parser.add_argument("planning_group_name", help="The name of the planning group for which your IKFast solution was generated")
+  parser.add_argument("kinematic_plugin_pkg", help="The name of the MoveIt! IKFast Kinematics Plugin to be updated")
+  parser.add_argument("base_link_name", help="The name of the base link that was used when generating your IKFast solution")
+  parser.add_argument("eef_link_name", help="The name of the end effector link that was used when generating your IKFast solution")
+  parser.add_argument("ikfast_output_path", help="The full path to the analytic IK solution output by IKFast")
+  parser.add_argument("--search_mode", default=search_modes[0], help="The search mode used to solve IK for robots with more than 6DOF")
+  parser.add_argument("--srdf_filename", help="The name of your robot. If your robot's srdf is not <robot_name>.srdf, then set this value with the correct filename. Defaults to <robot_name>.srdf")
+  parser.add_argument("--robot_name_in_srdf", help="The name of your robot as defined in the srdf. Defaults to <robot_name>")
+  parser.add_argument("--moveit_config_pkg", help="The robot moveit_config pagkacge. Defaults to <robot_name>_moveit_config")
+  return parser
+
+def populate_optional(args):
+  if args.srdf_filename is None:
+    args.srdf_filename = args.robot_name + ".srdf"
+  if args.robot_name_in_srdf is None:
+    args.robot_name_in_srdf = args.robot_name
+  if args.moveit_config_pkg is None:
+    args.moveit_config_pkg = args.robot_name + "_moveit_config"
+
+def print_args(args):
```

Comment:
is this function still helpful now that argparse is used?

Reply:
Some of the values are slightly changed so it's not exactly a 1-1. This is really useful for debugging.